lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
+    if !config.context.key?(context_key)
+      raise ArgumentError, "#{context_key} not found in Retriable.config.context"
+    end
+
+    retriable(config.context[context_key].merge(options), &block) if block
+  end
+
   def retriable(opts = {})
```

Comment:
I don't think i need the block here because we call yield here. `retriable_with_context` needs it because I need to pass the block into `#retriable` without calling it.